pkg/operator/operators/klusterlet/controllers/klusterletcontroller/klusterlet_controller.go

```diff
@@ -333,20 +335,39 @@ func getManagedKubeConfig(ctx context.Context, kubeClient kubernetes.Interface,
 	return helpers.LoadClientConfigFromSecret(managedKubeconfigSecret)
 }
 
-// ensureAgentNamespace create agent namespace if it is not exist
-func ensureAgentNamespace(ctx context.Context, kubeClient kubernetes.Interface, namespace string) error {
-	_, err := kubeClient.CoreV1().Namespaces().Get(ctx, namespace, metav1.GetOptions{})
+// ensureNamespaceWithAnnotation create the namespace if necessary and make sure it has the
+// given annotation
+func ensureNamespaceWithAnnotation(ctx context.Context, kubeClient kubernetes.Interface, namespace, annotationName,
+	annotationValue string) error {
+	ns, err := kubeClient.CoreV1().Namespaces().Get(ctx, namespace, metav1.GetOptions{})
```

Comment:
hrm, if it is not create only, maybe you can just use use applyNamespace in library go

Reply:
Use `applyNamespace` from the openshift library-go instead.